doc/crypto/api/ops/key-encapsulation.rst

```diff
@@ -0,0 +1,350 @@
+
+The original specification permitted a number of variants.
+The |API| uses the version specified in :cite-title:`SEC1`.
+
+The full ECIES scheme uses an elliptic-curve key agreement between the recipient's static public key and an ephemeral private key, to establish encryption and authentication keys for secure transmission of arbitrary-length messages to the recipient.
+
+An application using ECIES must select all of the following parameters:
+
+*   The elliptic curve for the initial key agreement.
+*   The KDF to derive the symmetric keys, and any label used in that derivation.
+*   The encryption and MAC algorithms.
+*   The additional data to include when computing the authentication.
+
+The |API| presents the key-agreement step of ECIES as a key-encapsulation algorithm.
+The key derivation, encryption, and authentication steps are left to the application.
+
+.. rationale::
+
+    Although it is possible to implement this in an application using key generation and key agreement, using the key-encapsulation functions enables an easy migration to other key-encapsulation algorithms, such as ML-KEM.
+
+.. admonition:: Implementation note
+
+    It is possible that some applications may need to use alternative versions of ECIES to interoperate with legacy systems.
+
+    While the application can implement this using key agreement functions, an implementation can choose to add these as a convenience with an :scterm:`implementation defined` key-encapsulation algorithm identifier.
+
+.. macro:: PSA_ALG_ECIES_SEC1
+    :definition: ((psa_algorithm_t)0x0c000100)
+
+    .. summary::
+        The Elliptic Curve Integrated Encryption Scheme (ECIES).
+
+    This key-encapsulation algorithm is defined by :cite-title:`SEC1` §5.1 under the name Elliptic Curve Integrated Encryption Scheme.
+
+    A call to `psa_encapsulate()` carries out steps 1 to 4 of the ECIES encryption process described in `[SEC1]` §5.1.3:
+
+    *   The elliptic curve to use is determined by the key.
+    *   The public key part of the input key is used as :math:`Q_V`.
+    *   Cofactor ECDH is used to perform the key agreement.
+    *   The shared secret :math:`Z` is output as the shared output key.
+    *   The ephemeral public key :math:`\overline{R}` is output as the encapsulation data.
+
+    A call to `psa_decapsulate()` carries out steps 2 to 5 of the ECIES decryption process described in `[SEC1]` §5.1.4:
+
+    *   The elliptic curve to use is determined by the key.
+    *   The encapsulation data is decoded as :math:`\overline{R}`.
+    *   The private key of the input key is used as :math:`d_V`.
+    *   Cofactor ECDH is used to perform the key agreement.
+    *   The shared secret :math:`Z` is output as the shared output key.
+
+    The encapsulation data produced by `PSA_ALG_ECIES_SEC1` is not authenticated.
+    When used in a full ECIES scheme, the authentication of the encrypted message implicitly confirms that the derived keys were identical.
```

Comment:
> the authentication of the encrypted message implicitly confirms that the derived keys were identical

Does it? Isn't this assuming that the authentication is key-committing, which is generally the case for hash-based MACs but not for other common authentication algorithms?

Reply:
So perhaps the wording is too specific about what the AE scheme, or cipher+MAC scheme ensures.

The gist of what I wanted to say was that explicit confirmation of the shared secret key is not used in ECIES. The scheme uses either a keyed MAC or authenticated encryption using a key derived from the shared secret to authenticate the encrypted message.

Reply:
I've reworded this to something more specific to ECIES and its purpose:

> In the full ECIES scheme, the authentication of the encrypted message using a key derived from the shared secret provides assurance that the message has not been manipulated.